x/emissions/module/abci.go

```diff
@@ -65,7 +65,7 @@ func EndBlocker(ctx context.Context, am AppModule) error {
 					sdkCtx.Logger().Warn(fmt.Sprintf("Error updating last inference ran: %s", err.Error()))
 				}
 				// Add Worker Nonces
-				nextNonce := types.Nonce{BlockHeight: blockHeight + topic.EpochLength}
+				nextNonce := types.Nonce{BlockHeight: blockHeight}
```

Comment:
Shouldn't we keep this as `blockHeight + topic.EpochLength`? Because this is called at end of block so means this nonce(block) was already passed without fullfilled. Wdyt?

Reply:
per [the ticket](https://uploads.linear.app/9b3a8ec3-68c2-4405-b1f3-51f697acd644/95a520f8-f890-417a-a448-ea3ec4bc04e5/ec7f533c-be30-48da-bea2-f492bf2ad739) the nonce refers to when the request was generated, and is the id for the entire sequence of events that cumulates in a network inference

Reply:
What @kpeluso said. 
Also, logs show that the request for nonce X is made at block X.